compiler/src/dotty/tools/dotc/core/Annotations.scala

```diff
@@ -57,15 +58,25 @@ object Annotations {
       val args = arguments
       if args.isEmpty then this
       else
+        // Checks if `tm` would result in any change by applying it to types
+        // inside the annotations' arguments and checking if the resulting types
+        // are different.
         val findDiff = new TreeAccumulator[Type]:
           def apply(x: Type, tree: Tree)(using Context): Type =
             if tm.isRange(x) then x
             else
               val tp1 = tm(tree.tpe)
-              foldOver(if tp1 frozen_=:= tree.tpe then x else tp1, tree)
+              foldOver(if tp1 == tree.tpe then x else tp1, tree)
```

Comment:
We basically never use `==` since it's not very well defined. What breaks if we would still use `frozen_=:=`?